perf_harness/src/main/java/io/opentelemetry/perf/OtlpPipelineDriver.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.perf;
+
+import eu.rekawek.toxiproxy.Proxy;
+import eu.rekawek.toxiproxy.ToxiproxyClient;
+import eu.rekawek.toxiproxy.model.Toxic;
+import eu.rekawek.toxiproxy.model.ToxicDirection;
+import eu.rekawek.toxiproxy.model.ToxicList;
+import eu.rekawek.toxiproxy.model.toxic.Latency;
+import io.opentelemetry.OpenTelemetry;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.exporters.otlp.OtlpGrpcSpanExporter;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.export.IntervalMetricReader;
+import io.opentelemetry.sdk.metrics.export.MetricExporter;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.Tracer;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+
+class OtlpPipelineDriver {
+
+  public static final String COLLECTOR_PROXY_PORT = "44444";
+
+  public static void main(String[] args) throws InterruptedException, IOException {
+    ToxiproxyClient toxiproxyClient = new ToxiproxyClient();
+    toxiproxyClient.reset();
+    Proxy collectorProxy = toxiproxyClient.getProxyOrNull("collector");
+
+    if (collectorProxy == null) {
+      collectorProxy =
+          toxiproxyClient.createProxy(
+              "collector", "0.0.0.0:" + COLLECTOR_PROXY_PORT, "otel-collector:55680");
+    }
+    collectorProxy.enable();
+
+    ToxicList toxics = collectorProxy.toxics();
+    Latency latency = toxics.latency("jittery_latency", ToxicDirection.DOWNSTREAM, 800);
+    latency.setJitter(10);
+
+    for (Toxic toxic : toxiproxyClient.getProxy("collector").toxics().getAll()) {
+      System.out.println("toxic = " + toxic.getName() + " : " + toxic.getToxicity());
+    }
+
+    IntervalMetricReader intervalMetricReader = setupSdk();
+
+    Tracer tracer = OpenTelemetry.getTracer("io.opentelemetry.perf");
+
+    for (int i = 0; i < 10000; i++) {
+      Span exampleSpan = tracer.spanBuilder("exampleSpan").startSpan();
+      try (Scope scope = tracer.withSpan(exampleSpan)) {
+        exampleSpan.setAttribute("good", "true");
```

Comment:
Maybe start using setup described in https://github.com/open-telemetry/opentelemetry-specification/pull/748 ?

Reply:
Yeah, I hadn't even begun to think about the shape of the data being pushed through here. My main goal with this PR was just to get something up and running, with toxiproxy in between the collector and the SDK, so we can start making it more interesting and realistic over time.